src/connectors/geo-search/connectGeoSearch.js

```diff
@@ -172,86 +172,97 @@ const connectGeoSearch = (renderFn, unmountFn = noop) => {
 
     let sendEvent;
 
-    const init = initArgs => {
-      const { state, helper, instantSearchInstance } = initArgs;
-      const isFirstRendering = true;
-
-      sendEvent = createSendEventForHits({
-        instantSearchInstance,
-        index: helper.getIndex(),
-        widgetType: $$type,
-      });
-
-      widgetState.internalToggleRefineOnMapMove = createInternalToggleRefinementOnMapMove(
-        noop,
-        initArgs
-      );
-
-      widgetState.internalSetMapMoveSinceLastRefine = createInternalSetMapMoveSinceLastRefine(
-        noop,
-        initArgs
-      );
-
-      renderFn(
-        {
-          items: [],
-          position: getPositionFromState(state),
-          currentRefinement: getCurrentRefinementFromState(state),
-          refine: refine(helper),
-          sendEvent,
-          clearMapRefinement: clearMapRefinement(helper),
-          isRefinedWithMap: isRefinedWithMap(state),
-          toggleRefineOnMapMove,
-          isRefineOnMapMove,
-          setMapMoveSinceLastRefine,
-          hasMapMoveSinceLastRefine,
-          widgetParams,
+    return {
+      $$type,
+
+      init(initArgs) {
+        const { helper, instantSearchInstance } = initArgs;
+        const isFirstRendering = true;
+
+        sendEvent = createSendEventForHits({
```

Comment:
In `index` widget, it runs `getRenderState` of all widgets before `init`. It means it will initially have `sendEvent: undefined`. I guess we're okay with that, since we call `init` right after that?

Reply:
I don't see how sendEvent could be defined before init since you need access to the InstantSearch. What did we do for the other sendEvent connectors @eunjae-lee ?

Reply:
I think your approach here makes sense. Before init, sendEvent means nothing.
I'm just reminded that I should change my previous implementations to this.

What I did before is to assign them inside getWidgetRenderState:

```js
getWidgetRenderState() {
  if (!sendEvent) {
    sendEvent = ...
  }

  return {
    ...
  }
}
```

We just need to make sure when we tests the return of `getWidgetRenderState` **BEFORE** init will have undefined sendEvent.

Reply:
I prefer that implementation, pushed the change

Reply:
hmm that's safer and especially matches the type, though. Okay, then let's go with it.